http/handler.go

```diff
@@ -185,6 +185,9 @@ func requestWrapTTL(r *http.Request, req *logical.Request) (*logical.Request, er
 		}
 		req.WrapTTL = time.Duration(seconds) * time.Second
 	}
+	if int64(req.WrapTTL) < 0 {
```

Comment:
`req.WrapTTL` is a `time.Duration` which specializes `int64`, so this keeps precision. It's only being used for a check here. (I think you can actually just do the comparison directly, but this makes it obvious what's happening.)